services/src/main/scala/cromwell/services/metadata/impl/deleter/DeleteMetadataActor.scala

```diff
@@ -0,0 +1,85 @@
+package cromwell.services.metadata.impl.deleter
+
+import java.time.OffsetDateTime
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import akka.util.Timeout
+import common.util.StringUtil.EnhancedToStringable
+import cromwell.core.WorkflowId
+import cromwell.services.MetadataServicesStore
+import cromwell.services.instrumentation.CromwellInstrumentation
+import cromwell.services.metadata.MetadataArchiveStatus
+import cromwell.services.metadata.MetadataArchiveStatus.{Archived, ArchivedAndPurged}
+import cromwell.services.metadata.impl.MetadataDatabaseAccess
+import cromwell.services.metadata.impl.deleter.DeleteMetadataActor._
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+class DeleteMetadataActor(deleteMetadataConfig: DeleteMetadataConfig,
+                          override val serviceRegistryActor: ActorRef)
+  extends Actor
+  with ActorLogging
+  with MetadataDatabaseAccess
+  with MetadataServicesStore
+  with CromwellInstrumentation {
+
+  implicit val ec: ExecutionContext = context.dispatcher
+  implicit val askTimeout: Timeout = new Timeout(60.seconds)
+
+  // Send an initial delete message to get things started:
+  self ! DeleteNextWorkflowMessage
```

Comment:
Could it be a good idea to schedule this on a timer so it happens not-immediately, so the system has a chance to stabilize after starting up?

Reply:
Per a mob discussion on the `archiver` side, we decided to keep this as a simple message for now, and consider a delay if and when we see early starts causing issues.